### ISSUE TITLE
make `io` package into `dataframe` subpackage, add vak.DataFrame class

### ISSUE THREAD

**Issue by NickleDave** [vocalpy/vak#205]
currently all modules in `io` are really about creating a `dataframe` but this is not super clear to a naive reader

To make this more clear, rename the sub-package `dataframe`, and add a class `DataFrame` that sub-classes.

This would also enable using the `DataFrame` class throughout the codebase to make other functions more readable and concise,
e.g. we could call `vak_df.has_unlabeled()` method instead of needing to import the `labels` module and pass the dataframe to the function `labels.has_unlabeled(vak_df)`.

the `dataframe.from_files` function is essentially a high-level class method.
the `spect.to_dataframe` is also a class method. It could be renamed `from_spect_files`.

similarly could call `get_labeled_timebins`, `get_annotation`, etc.

so:
keep `dataframe` subpackage but add class that subclasses pandas.DataFrame
split up into modules in such a way that the class does not occupy 10k lines.
One possiblity: 
```
vak/
  dataframe/
    class.py
    methods.py
```
where `methods.py` contains class methods that are imported and used by `class.py` -- not sure if that would work though? It's wouldn't really a class method in the standard sense, because we don't use `cls`

**Comment by NickleDave** [comment]
thinking it should be like
```
vak
├── dataframe
│   ├── class.py  # subclass `pandas.Dataframe`, add `classmethod` that uses `dataframe` / `functional` function
│   ├── functional.py  # currently `dataframe`
├── io
│   ├── spect.py
│   ├── audio.py
```

i.e., factor out `dataframe` and put in its own sub-package, but keep an `io` sub-package --- may make sense to move other things into that `io` sub-package, e.g. `annotation`, `files`

**Comment by NickleDave** [comment]
one issue related to this is that currently we assume the `DataFrame` will always have a `spect_path` column.
There's no way to create a `DataFrame` with just audio files.

The current `DataFrame` is "defined" (very roughly) in `vak.io.spect.to_dataframe` by just a constant declaring the columns

And so `vak.io.dataframe.from_files` is basically a high-level wrapper that first calls `vak.io.audio.to_spect` on a bunch of audio files to convert them to spectrogram files **or** just takes a bunch of pre-existing spectrogram files, then finally calls `vak.io.spect.to_dataframe`

this is still convenient functionality to have ... I think?
not sure how to refactor in a way that keeps it while still making it possible to have "just audio" dataframes

**Comment by NickleDave** [comment]
Closing because this kind of refactor will be addressed by #558
